packages/e2e-tests/config/setup-test-framework.js

```diff
@@ -187,28 +187,34 @@ async function simulateAdverseConditions() {
 		return;
 	}
 
-	const client = await page.target().createCDPSession();
+	if ( OFFLINE ) {
+		await page.setOfflineMode( true );
+		await page.emulateNetworkConditions( {
+			// Download speed (bytes/s)
+			download: -1,
+			// Upload speed (bytes/s)
+			upload: -1,
+			// Latency (ms)
+			latency: 0,
```

Comment:
Docs aren't very clear on this either, it says -1 to disable download/upload, and 0 to disable latency. But does 'disable' mean 'offline'?

It works when tested 🤷‍♂️

Reply:
I feel like disable probably means no network condition emulation:
https://github.com/puppeteer/puppeteer/blob/v10.1.0/docs/api.md#pageemulatenetworkconditionsnetworkconditions

So maybe `setOfflineMode` is all that's needed?

I've asked for clarification:
https://github.com/puppeteer/puppeteer/pull/6759/files#r669263455

Reply:
Yeah, I think `setOfflineMode` is all we need. [Under the hood](https://github.com/puppeteer/puppeteer/blob/dd470c7a226a8422a938a7b0fffa58ffc6b78512/src/common/NetworkManager.ts#L192), it just sets the `offline` key to `true`.

Reply:
Cool, I've switched to just using that 👍